src/plugins/ui_feature/ui_sheet.ts

```diff
@@ -272,4 +262,31 @@ export class SheetUIPlugin extends UIPlugin {
     splitWord.push(wordPart);
     return splitWord;
   }
+
+  /**
+   * Check that any "sheetId" in the command matches an existing
+   * sheet.
+   */
+  private checkSheetExists(cmd: Command): CommandResult {
+    if ("sheetId" in cmd && this.getters.tryGetSheet(cmd.sheetId) === undefined) {
+      return CommandResult.InvalidSheetId;
+    }
+    return CommandResult.Success;
+  }
+
+  /**
+   * Check if zones in the command are well formed and
+   * not outside the sheet.
+   */
+  private checkZones(cmd: Command): CommandResult {
+    const sheetId = "sheetId" in cmd ? cmd.sheetId : this.getters.tryGetActiveSheetId();
+    const zones = this.getters.getCommandZones(cmd);
+    if (!sheetId && zones.length > 0) {
+      throw new Error("Trying to dispatch an active sheet dependent command before initialization");
```

Comment:
shouldn't it return a non-successful CommandResult instead of throwing ? Also, just by reading this, I'm not sure what it has to do with initialization